### PR TITLE
feat: Refactor Cocoa into Apple

### PR DESCRIPTION
**Pull request by dcramer**
- cocoa => apple
- add guides for ios, macos, watchos, tvos
- add various setup snippets
- merge swift/objc code
- remove old version instructions
- suggest session tracking by default

### CODE REVIEW

File: src/components/basePage.tsx
Comment (dcramer) [inline src/components/basePage.tsx:82]: this fixed an issue where if there was no toc we were pushing the sdk info to the footer

File: src/includes/configuration/auto-session-tracking/apple.mdx
Comment (bruno-garcia) [inline src/includes/configuration/auto-session-tracking/apple.mdx:5]: Prefer the typed overload from the map:
```suggestion
SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.enableAutoSessionTracking = true
}
```

File: src/includes/configuration/session-tracking-interval/apple.mdx
Comment (bruno-garcia) [inline src/includes/configuration/session-tracking-interval/apple.mdx:6]: Prefer the typed overload from the map:
```suggestion
SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.enableAutoSessionTracking = true
    options.sessionTrackingIntervalMillis = 60000
}
```

File: src/includes/getting-started-config/apple.mdx
Comment (bruno-garcia) [inline src/includes/getting-started-config/apple.mdx:14]:
```suggestion
        options.enableAutoSessionTracking = true // Enable session tracking
```

File: src/includes/set-release/apple.mdx
Comment (bruno-garcia) [inline src/includes/set-release/apple.mdx:7]: Prefer the typed overload from the map:
```suggestion
SentrySDK.start { options in
    options.dsn = "___PUBLIC_DSN___"
    options.releaseName = "my-project-name@2.3.12"
}
```

File: src/includes/set-release/apple.mdx
Comment (bruno-garcia) [inline src/includes/set-release/apple.mdx:13]: String key is `release`:

https://github.com/getsentry/sentry-cocoa/blob/6fc3fb537c99b8d61854c22928abd96bf843a402/Sources/Sentry/SentryOptions.m#L109-L111

```suggestion
    @"release": @("my-project-name@2.3.12")
```

File: src/wizard/apple/index.md
Comment (EvanPurkhiser) [inline src/wizard/apple/index.md:1]: Make sure to update this in sentry's `platformCategories`

File: src/includes/getting-started-config/apple.mdx
Comment (PeloWriter) [inline src/includes/getting-started-config/apple.mdx:1]:
```suggestion
Initialize the SDK as soon as possible in your application lifecycle, such as in your AppDelegate `application:didFinishLaunchingWithOptions` method:
```

**Comment by untitaker** [comment]
#### 📝Changed routes:


| [**/**](https://sentry-docs-git-feat-refactor-cocoa.sentry.dev/) |
|:-:|
|<a href="https://sentry-docs-git-feat-refactor-cocoa.sentry.dev/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/E7FJAw5FRSRzgGvgpNPdPt0yIbI_.png">(view full size)</a> |







Commit 8fa72de4071bf5ff1be80ee23cda9e57c41f28a7 (https://sentry-docs-git-feat-refactor-cocoa.sentry.dev).

File: src/wizard/apple/index.md
Comment (PeloWriter) [inline src/wizard/apple/index.md:2]: Not sure if Apple is the right name here, but I defer to others' expertise. 

```suggestion
name: iOS/OS (Cocoa)
```

File: src/platforms/apple/common/install/swift-package-manager.mdx
Comment (bruno-garcia) [inline src/platforms/apple/common/install/swift-package-manager.mdx:7]: I had this problem trying to use Sentry: https://github.com/bruno-garcia/SentryCocoaLumberjack/blob/e30623b76a27647e87d8cba0752724b782ec40ab/Package.swift#L13

can we replace `version` below with the latest from release registry magically here?

```suggestion

When using SPM version 5.2 or later, you must specify the the target name too:

```swift
.package(name: "Sentry", url: "https://github.com/getsentry/sentry-cocoa", from: "version"),
```

<Note><markdown>
```

**Review by PeloWriter — APPROVED** [review]
One q on the use of Apple, and one language nit. But this moves us forward. Thank you

File: src/platforms/apple/guides/watchos/index.mdx
Comment (bruno-garcia) [inline src/platforms/apple/guides/watchos/index.mdx:39]: This page should describe what limitations we have with watchOS symbolication, and how those affect all crash reporting tools and not only Sentry.
I don't know it well enough to write this well. @mitsuhiko @HazAT ?

File: src/wizard/apple/index.md
Comment (dcramer) [reply src/wizard/apple/index.md:2]: theres already an iOS page so im not super concerned about that. "OS" doesnt really make a ton of ssense here since we're only show apple/cocoa

Comment (bruno-garcia) [reply src/wizard/apple/index.md:2]: If listing actual platforms, consider we have:

iOS, watchOS, tvOS, iPadOS.. macOS

Comment (bruno-garcia) [reply src/wizard/apple/index.md:2]: I wonder if Apple announced a new one today

Comment (dcramer) [reply src/wizard/apple/index.md:2]: they announced new date for these and i might have bought a watch

Comment (bruno-garcia) [reply src/wizard/apple/index.md:2]: LOL! Last time we talked in person, I bought one too.
Good buy

File: src/wizard/apple/index.md
Comment (dcramer) [reply src/wizard/apple/index.md:1]: https://github.com/getsentry/sentry/pull/20799

**Comment by dcramer** [comment]
@HazAT will wait for your feedback since this is a large change

**Review by bruno-garcia — APPROVED** [review]
@philipphofmann and @HazAT to have the last call but LGTM

File: src/platforms/apple/guides/watchos/index.mdx
Comment (bruno-garcia) [reply src/platforms/apple/guides/watchos/index.mdx:39]: https://app.asana.com/0/1162483910998154/1185636138279961

Some note was added here: https://github.com/getsentry/sentry-docs/pull/1924/files
Is it lost?

Would be nice to have a better technical explanation what this means

**Review by philipphofmann — COMMENTED** [review]
In fact, I'm not sure if this is a good idea. Cocoa is a combination of frameworks that you use to develop applications for macOS, iOS, iPadOS, tvOS, and watchOS. Our Cocoa SDK only works if you use Cocoa in your application. You can also build apps for Apple`s platforms that don't run Cocoa. If you use Java, Flutter, etc. to build an app that runs on one of these platforms you can't use our Cocoa SDK for that. @HazAT what do you think?

I'm working on a PR for adding code samples for configuration, enriching error data, and such. Just for you to know @dcramer so we don't end up doing the same stuff twice.

**Comment by HazAT** [comment]
@philipphofmann I think we should focus on what's obvious from the outside while technically not being 100% correct. So I feel like, strictly looking at the technical side the `cocoa` -> `apple` rename is not perfect but in terms of discoverability and understandability I think it's better.

**Review by HazAT — APPROVED** [review]
Good first step 👍 
Still missing a few details that we'll add later on

**Comment by philipphofmann** [comment]
> @philipphofmann I think we should focus on what's obvious from the outside while technically not being 100% correct. So I feel like, strictly looking at the technical side the `cocoa` -> `apple` rename is not perfect but in terms of discoverability and understandability I think it's better.

I already raised my concerns and I don't have a strict opinion on this. As most people like this refactoring, I'm fine with it.

File: src/wizard/apple/index.md
Comment (dcramer) [reply src/wizard/apple/index.md:27]: This gets embedded in the product and doesnt have as much capability, so by adding objective c here we effectively double the length of instructions. If its 50% of people use objc we can do that, if its 90%, then we remove swift, if its 10%, then they can go to full docs to figure it out.

**Comment by dcramer** [comment]
I'm ok with keeping this Cocoa, but no one cares if its Cocoa or Apple, they just wanna know how to use Sentry on iOS, and I generally agree w/ the sentiment that bucketing it as "Cocoa" has never felt obvious to end users. It's about as obvious a bucketing it as "objective c" is.

**Comment by dcramer** [comment]
https://twitter.com/mitsuhiko/status/1306135791094181888

It seems like Apple may not be the clear cut solution, but its probably the at-least-slightly-better-than-other-options from discoverability.

File: src/wizard/apple/index.md
Comment (dcramer) [reply src/wizard/apple/index.md:27]: (I made an assumption here to be clear, I'm ok changing it, but we should keep the onboarding wizard instructions as short and sweet as possible and funnel them to the deeper docs as needed)

File: src/docs/product/releases/health.mdx
Comment (dcramer) [reply src/docs/product/releases/health.mdx:66]: Yeah lets do this in a follow-up as its likely there is more cleanup needed than just that around release health.

**Comment by dcramer** [comment]
One other followup: we likely should expand the index page of this, possibly hiding a bunch of common content? and adding additional detials about multi-platform build, non-cocoa options, etc

**Comment by dcramer** [comment]
I'm going to ship in interest if iteration and moving forward (and moving forward w/ @philipphofmann's other PR). I'd encourage us to not think of this as locked in and if we come up with something that's a better UX let's move on it.

**Comment by philipphofmann** [comment]
> https://twitter.com/mitsuhiko/status/1306135791094181888
> 
> It seems like Apple may not be the clear cut solution, but its probably the at-least-slightly-better-than-other-options from discoverability.

Thanks for pointing it out @dcramer and thanks for the PR 👏

**Pull request merged**
